aten/src/ATen/core/ivalue_inl.h

```diff
@@ -664,7 +637,32 @@ struct C10_EXPORT ivalue::Object final : c10::intrusive_ptr_target {
 // see concrete implementation in python_ivalue.h
 struct ivalue::PyObjectHolder : c10::intrusive_ptr_target {
  public:
+  struct InferredType {
```

Comment:
this seems invalid to be put inside PyObjectHolder, as it's a generic type used by any cases (might be excluding PyObjectHolder sometimes). Can we maybe put this into `jit_type.h` instead?